src/battle/overlay_12_0224E4FC.c

```diff
@@ -5541,3 +5542,842 @@ BOOL TryFling(BattleSystem *bsys, BATTLECONTEXT *ctx, int battlerId) {
+        }
+    } else {
+        ctx->battleMons[ctx->battlerIdAttacker].unk88.metronomeTurns = 0;
+    }
+}
+
+//Related to send out Pokemon crys..?
+int ov12_02256748(BATTLECONTEXT *ctx, int battlerId, int battlerType, BOOL encounter) {
+    int ret;
+    int color;
+    BOOL half;
+    
+    if (encounter == TRUE && (battlerType == 2 || battlerType == 3)) {
+        half = TRUE;
+    } else {
+        half = FALSE;
+    }
+
+    ret = 0;
+    
+    if (half == TRUE) {
+        ret = 0;
+    }
+    
+    color = sub_020880B0(ctx->battleMons[battlerId].hp, ctx->battleMons[battlerId].maxHp, 48);
+    
+    if ((ctx->battleMons[battlerId].status & STATUS_ALL) || (color != 4 && color != 3)) {
+        ret = 11;
+    }
+
+    return ret;
+}
+
+BOOL Battler_CanSelectAction(BATTLECONTEXT *ctx, int battlerId) {
+    BOOL ret = TRUE;
+    
+    if ((ctx->battleMons[battlerId].status2 & STATUS2_RECHARGE) ||
+        (ctx->battleMons[battlerId].status2 & STATUS2_RAGE) ||
+        (ctx->battleMons[battlerId].status2 & STATUS2_UPROAR) ||
+        (ctx->battleMons[battlerId].status2 & STATUS2_LOCKED_INTO_MOVE)) {
+        ret = FALSE;
+    }
+    
+    return ret;
+}
+
+void ov12_022567D4(BattleSystem *bsys, BATTLECONTEXT *ctx, Pokemon *mon) {
+    PlayerProfile *profile = BattleSystem_GetPlayerProfile(bsys, BATTLER_PLAYER);
+    int a3 = ov12_0223AB60(bsys);
+    int terrain = BattleSystem_GetTerrainId(bsys);
+    int ballId;
+    
+    if (BattleSystem_GetBattleType(bsys) & BATTLE_TYPE_9) {
+        ballId = BallToItemId(BattleSystem_GetMonBall(bsys, mon));
+    } else {
+        ballId = ctx->itemTemp;
+    }
+    
+    sub_020720FC(mon, profile, ballId, a3, terrain, HEAP_ID_BATTLE);
+}
+
+u8 ov12_0225682C(BATTLECONTEXT *ctx, int battlerId) {
+    return ctx->unk_2300[battlerId][0];
+}
+
+BOOL BattlerCheckSubstitute(BATTLECONTEXT *ctx, int battlerId) {
+    BOOL ret = FALSE;
+    
+    if (ctx->selfTurnData[battlerId].unk14 & (1 << 3)) {
+        ret = TRUE;
+    }
+    
+    return ret;
+}
+
+BOOL ov12_02256854(BattleSystem *bsys, BATTLECONTEXT *ctx) {
+    PlayerProfile *profile = BattleSystem_GetPlayerProfile(bsys, BATTLER_PLAYER);
+    u32 trainerId = PlayerProfile_GetTrainerID(profile);
+    u32 gender = PlayerProfile_GetTrainerGender(profile);
+    const u16 *name = PlayerProfile_GetNamePtr(profile);
+    
+    if (trainerId == ctx->battleMons[ctx->battlerIdAttacker].otid &&
+        gender == ctx->battleMons[ctx->battlerIdAttacker].metGender &&
+        !StringNotEqualN(name, &ctx->battleMons[ctx->battlerIdAttacker].otName[0], 7)) {
+        return TRUE;
+    }
+    
+    return FALSE;
+}
+
+BOOL ov12_022568B0(BattleSystem *bsys, Pokemon *mon) {
+    PlayerProfile *profile = BattleSystem_GetPlayerProfile(bsys, BATTLER_PLAYER);
+    u32 trainerId = PlayerProfile_GetTrainerID(profile);
+    u32 gender = PlayerProfile_GetTrainerGender(profile);
+    const u16 *name = PlayerProfile_GetNamePtr(profile);
+    const u16 otName[8];
```

Comment:
length constant

Reply:
still needs a length constant here

Reply:
me when I forget to hit cntrl + s on a file before pushing

Reply:
still missing a length constant....

Reply:
Resolving since going into the code editor shows the change occurred, so this might just be an issue with github's difference finding system or smth